### ISSUE TITLE
wrong_self_convention: false positive if method name is the same as prefix

### ISSUE THREAD

**Issue by dtolnay** [Manishearth/rust-clippy#1285]
For example [`aster::item::ItemUsePathBuilder::as_`](https://docs.rs/aster/0.30.0/aster/item/struct.ItemUsePathBuilder.html#method.as_).

```rust
// use std::vec::Vec as MyVec;
let item = builder.item().use_()
    .ids(&["std", "vec", "Vec"]).build()
    .as_("MyVec");
```

The method should be called `"as"` but that is not allowed so `"as_"` is the most readable alternative. Clippy should only enforce self convention if there is something after "as_".

**Comment by oli-obk** [comment]
Haha my bad... That's what happens when the stdlib's iterators don't fit the use case and one iterates manually.... One forgets the zero iteration case

**Comment by devonhollowood** [comment]
I think I'll take a shot at this if that's alright.

**Comment by Manishearth** [comment]
Just to be clear, you don't need to ask before working on something. Leaving a comment saying that you are working on it is good to have, but unnecessary.
